### DISCUSSION TITLE
Mesh network oscillating between 1 and 2 partitions

### DISCUSSION THREAD

**Discussion by fgervais** [openthread/openthread#7862] [Q&A] [answered]
I have 2 border routers on the 2 opposite side of my home.

![image](https://user-images.githubusercontent.com/5411996/176927657-d0f1e010-dfa9-40c0-8de0-bc2be582e180.png)

Most of the time, they don't really see each other. They see the MLE advertisements but it's below `OPENTHREAD_CONFIG_MLE_PARTITION_MERGE_MARGIN_MIN` so they split in 2 partitions and everything is good.

However, I guess that the MLE link margin is just below `OPENTHREAD_CONFIG_MLE_PARTITION_MERGE_MARGIN_MIN` because at some point they feel like merging partitions, one of them becomes child, drops childs etc etc.

The devices associated with this BR loose network access for a little while.

Then short after they figure out that merging was a bad idea and the split again. Again, device stops responding for a little while.

This is only about 12 hours on one of the BRs:

```
counters mle
Role Disabled: 0
Role Detached: 43
Role Child: 22
Role Router: 0
Role Leader: 21
Attach Attempts: 21
Partition Id Changes: 43
Better Partition Attach Attempts: 330
Parent Changes: 0
```

I don't really understand because I feel like `OPENTHREAD_CONFIG_MLE_PARTITION_MERGE_MARGIN_MIN` is there for this exact use-case. 

What could make my case special for it to fail?

Is my best solution to bump `OPENTHREAD_CONFIG_MLE_PARTITION_MERGE_MARGIN_MIN` to 20 and rebuild?

**Comment by jwhui** [comment, marked as answer]
Increasing `OPENTHREAD_CONFIG_MLE_PARTITION_MERGE_MARGIN_MIN` to increase the hysteresis range should help.

Separately, Thread-over-Infrastructure (TREL) allows BRs to incorporate infrastructure links (i.e. Wi-Fi or Ethernet) into the Thread topology and is intended to directly address this situation. It basically allows BRs to transport IEEE 802.15.4 frames over UDP/IP. If you like, you can try it out.

You can enable TREL in your build by setting the [`OTBR_TREL` build option](https://github.com/openthread/ot-br-posix/blob/6ae74531dfbdee6fac0052eb110d2bf53a5cd5a3/etc/cmake/options.cmake#L82).

You can see if TREL is enabled in your build with the [`trel`](https://github.com/openthread/openthread/tree/main/src/cli#trel) CLI command.

**Reply by fgervais** [reply]
Excellent, I will look at TREL in more details and give it a try.

As usual, thank you for taking the time to respond, much appreciated.
